### PR TITLE
feat: Conversation roles

### PR DESCRIPTION
**Pull request by AndyLnd**

### CODE REVIEW

File: src/script/components/panel/userDetails.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+ *
+ */
+
+import {ParticipantAvatar} from 'Components/participantAvatar';
+import {User} from '../../entity/User';
+
+import 'Components/availabilityState';
+
+interface UserDetailsProps {
+  participant: User;
```
Comment (ffflorian) [inline src/script/components/panel/userDetails.ts]:
```suggestion
  participant: ko.observable<User>;
```

File: src/script/components/panel/userDetails.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  isVerified?: boolean;
```
Comment (ffflorian) [inline src/script/components/panel/userDetails.ts]:
```suggestion
  isVerified?: ko.PureComputed<boolean>;
```

File: src/script/components/userList.ts [hunk 3]
```diff
@@ -170,5 +230,45 @@ ko.components.register('user-list', {
         viewportObserver.removeElement(this.lazyTriggerElement);
       }
     };
+
+    this.memberUsers = ko.observable<User[]>([]);
+    this.adminUsers = ko.observable<User[]>([]);
+    this.users = ko.observable<User[]>([]);
```
Comment (ffflorian) [inline src/script/components/userList.ts:236]: Why not use an `observableArray`?

File: src/script/team/TeamRepository.js [hunk 4]
```diff
@@ -94,24 +90,19 @@ export class TeamRepository {
...
-        if (teamData) {
-          const teamEntity = this.teamMapper.mapTeamFromObject(teamData);
-          this.team(teamEntity);
-          return this.updateTeamMembers(teamEntity);
-        }
-
-        this.team(new TeamEntity());
-      })
-      .then(() => {
-        // doesn't need to be awaited because it publishes the account info over amplify.
```
Comment (ffflorian) [inline src/script/team/TeamRepository.js:110]: Please don't remove this info text.

File: src/script/view_model/panel/BasePanelViewModel.ts [hunk 5]
```diff
@@ -17,8 +17,31 @@
  *
  */
 
+import ko from 'knockout';
+import {ConversationRepository} from '../../conversation/ConversationRepository';
```
Comment (ffflorian) [inline src/script/view_model/panel/BasePanelViewModel.ts:22]:
```suggestion

import {ConversationRepository} from '../../conversation/ConversationRepository';
```

File: src/script/components/panel/userDetails.ts [hunk 6]
```diff
@@ -0,0 +1,89 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
```
Comment (lipis) [inline src/script/components/panel/userDetails.ts]:
```suggestion
 * Copyright (C) 2019 Wire Swiss GmbH
```

File: src/script/components/userList.ts (hunk 3 above)
Comment (AndyLnd) [reply src/script/components/userList.ts:236]: It doesn't make any difference.

File: src/script/view_model/panel/ConversationParticipantsViewModel.ts [hunk 7]
```diff
@@ -17,11 +17,23 @@
...
+
+import {sortByPriority} from 'Util/StringUtil';
+
+import {User} from '../../entity/User';
 import {MotionDuration} from '../../motion/MotionDuration';
+import {BasePanelViewModel, PanelViewModelProps} from './BasePanelViewModel';
 
 export class ConversationParticipantsViewModel extends BasePanelViewModel {
-  constructor(params) {
+  searchRepository: any;
```
Comment (Yserz) [inline src/script/view_model/panel/ConversationParticipantsViewModel.ts]: Let's try to get rid of the any's

**Pull request merged**